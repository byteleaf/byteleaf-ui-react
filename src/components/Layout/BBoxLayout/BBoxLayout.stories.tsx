import React from 'react';
import Typography from '@material-ui/core/Typography';
import { BBoxLayout } from './BBoxLayout';

export default {
  title: 'Layout|BBoxLayout',
};

export const BBoxLayoutStory = () => (
  <BBoxLayout>
    <div style={{ backgroundColor: '#ddd', padding: '8px' }}>
      <Typography>This div is in a boxed layout.</Typography>
    </div>
  </BBoxLayout>
);

BBoxLayoutStory.story = {
  name: 'Default',
  parameters: {
    notes: 'Standard boxed layout. Will use half of the screen on large displays.',
  },
};
