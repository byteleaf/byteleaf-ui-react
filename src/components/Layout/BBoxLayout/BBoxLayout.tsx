import React from 'react';
import { Grid } from '@material-ui/core';

export const BBoxLayout = ({ children }: { children: JSX.Element | JSX.Element[] }) => (
  <Grid container justify="center">
    <Grid item xs={12} sm={12} md={8} lg={6}>
      {children}
    </Grid>
  </Grid>
);
