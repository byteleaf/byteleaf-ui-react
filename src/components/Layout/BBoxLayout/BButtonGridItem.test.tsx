import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BBoxLayout } from './BBoxLayout';

describe('BBoxLayout', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BBoxLayout>
        <div>Layout</div>
      </BBoxLayout>,
    );

    expect(container).toMatchSnapshot();
  });
});
