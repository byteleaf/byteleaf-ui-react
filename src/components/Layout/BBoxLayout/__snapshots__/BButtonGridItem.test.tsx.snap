// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BBoxLayout renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6"
    >
      <div>
        Layout
      </div>
    </div>
  </div>
</div>
`;
